SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -227,4 +227,14 @@ Trace_SpanId SpatialEventTracer::GetSpanId(const EntityComponentId& Id) const
 
 		return *SpanId;
 }
+FString SpatialEventTracer::SpanIdToString(const Trace_SpanId& SpanId)
+{
+	FString HexStr;
+	for (int i = 0; i < 16; i++)
+	{
+		HexStr += FString::Printf(TEXT("%02x"), SpanId.data[i]);
```

Comment:
Another option is %02x%02x%02x%02x to avoid cycling strings

Reply:
Or taking advantage of %02x takes an unsigned integer which should cover at least 4 SpanId elements (needs to be 0-padded for higher widths)

Reply:
We could just avoid this in critical paths though

Reply:
Sorry, what is the suggested change here?

Reply:
```suggestion
		TCHAR Buffer[3];
		if (TCString<TCHAR>::Snprintf(Buffer, sizeof(Buffer), TEXT("%02X"), SpanId.data[i]) > 0)
		{
			HexStr += Buffer;
		}
```

Reply:
Will make this change in the user event tracing PR.